### PR TITLE
MYST-682 Move mysterium_server logic to "mysterium_node service" subcommand

### PR DESCRIPTION
**Pull request by Waldz**
```
Output of `bin/build && bin/run help service`


NAME:
   mysterium_node service - Starts and publishes service on Mysterium Network

USAGE:
   mysterium_node service [command options]

OPTIONS:
   --identity value             Keystore's identity used to provide service. If not given identity will be created automatically
   --identity.passphrase value  Used to unlock keystore's identity
   --openvpn.protocol value     Openvpn protocol to use. Options: { udp, tcp } (default: "udp")
   --openvpn.port value         Openvpn port to use. Default 1194 (default: 1194)
   --location.country value     Service location country. If not given country is autodetected
```

### CODE REVIEW

File: cmd/commands/service/command.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+				ctx.String("openvpn.proto"),
+				ctx.Int("openvpn.port"),
+
+				ctx.String("location.country"),
+			}
+			serviceManager = service.NewManager(nodeOptions, serviceOptions)
+
+			cmd.RegisterSignalCallback(utils.SoftKiller(stopCommand))
+
+			go nodeInstance.Start()
```
Comment (soffokl) [inline cmd/commands/service/command.go]: `nodeInstance.Start()` returns an error, should we at least log it somehow?

File: communication/nats/discovery/address_test.go [hunk 2]
```diff
@@ -95,6 +95,20 @@ func TestNewAddressForContact_UnknownDefinition(t *testing.T) {
 	assert.Nil(t, address)
 }
 
+func TestAddress_Disconnect_EmptyAddress(t *testing.T) {
+	address := &AddressNATS{}
+	address.Disconnect()
```
Comment (soffokl) [inline communication/nats/discovery/address_test.go]: Do we need any assertion here, or it checks "no panic" case?

File: core/service/factory.go [hunk 3]
```diff
@@ -100,55 +108,48 @@ func NewCommand(options CommandOptions) *Command {
...
 		},
 		checkOpenvpn: func() error {
-			return openvpn.CheckOpenvpnBinary(options.OpenvpnBinary)
+			return openvpn.CheckOpenvpnBinary(nodeOptions.OpenvpnBinary)
 		},
-		checkDirectories: options.Directories.Check,
+		checkDirectories: nodeOptions.Directories.Check,
 		openvpnServiceAddress: func(outboundIP, publicIP string) string {
-			//TODO public ip could be overriden by arg options if needed
+			//TODO public ip could be overriden by arg nodeOptions if needed
```
Comment (soffokl) [inline core/service/factory.go]: overriden -> overridden

File: core/service/factory.go [hunk 4]
```diff
@@ -100,55 +108,48 @@ func NewCommand(options CommandOptions) *Command {
...
 			if publicIP != outboundIP {
-				forwardInfo := fmt.Sprintf("%s:%v -> %s:%v", publicIP, options.OpenvpnPort, outboundIP, options.OpenvpnPort)
 				log.Warnf(
 					`WARNING: It seems that publicaly visible ip: [%s] does not match your local machines ip: [%s].
```
Comment (soffokl) [inline core/service/factory.go]: publicaly -> publicly

File: core/service/factory.go [hunk 5]
```diff
@@ -100,55 +108,48 @@ func NewCommand(options CommandOptions) *Command {
...
+		checkDirectories: nodeOptions.Directories.Check,
 		openvpnServiceAddress: func(outboundIP, publicIP string) string {
-			//TODO public ip could be overriden by arg options if needed
+			//TODO public ip could be overriden by arg nodeOptions if needed
 			if publicIP != outboundIP {
-				forwardInfo := fmt.Sprintf("%s:%v -> %s:%v", publicIP, options.OpenvpnPort, outboundIP, options.OpenvpnPort)
 				log.Warnf(
 					`WARNING: It seems that publicaly visible ip: [%s] does not match your local machines ip: [%s]. 
-You should probaly need to do port forwarding on your router: %s.`,
+You should probaly need to do port forwarding on your router: %s:%v -> %s:%v.`,
```
Comment (soffokl) [inline core/service/factory.go]: probaly -> probably

File: cmd/commands/service/command.go [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+)
+
+// NewCommand function creates service command
+func NewCommand() *cli.Command {
+	var nodeInstance *node.Node
+	var serviceManager *service.Manager
+
+	stopCommand := func() error {
+		if err := serviceManager.Kill(); err != nil {
+			return err
```
Comment (tadovas) [inline cmd/commands/service/command.go]: Maybe all errors on stop should be logged as warnings, and stop should proceed anyway? Now return in the middle of stop leaves command in interesting state

File: cmd/mysterium_server/mysterium_server.go [hunk 7]
```diff
@@ -58,22 +57,128 @@ func main() {
...
+	flags := flag.NewFlagSet(args[0], flag.ContinueOnError)
+
+	err = cmd.ParseDirectoryArguments(flags, &options.NodeOptions.Directories)
+	if err != nil {
+		return
+	}
+
+	flags.StringVar(
+		&options.NodeOptions.OpenvpnBinary,
+		"openvpn.binary",
```
Comment (interro) [inline cmd/mysterium_server/mysterium_server.go]: Do `mysterium_node service` accepts` openvpn.binary` parameter?

File: cmd/commands/service/command.go (hunk 6 above)
Comment (Waldz) [reply cmd/commands/service/command.go]: Improved

File: cmd/mysterium_server/mysterium_server.go (hunk 7 above)
Comment (Waldz) [reply cmd/mysterium_server/mysterium_server.go]: Yes, it's passed in global variables:
```
mysterium_node --openvpn.binary=openvpn service --openvpn.protocol=tcp --openvpn.port=1111
```

**Pull request merged**